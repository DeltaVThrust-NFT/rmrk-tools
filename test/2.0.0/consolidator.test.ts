import { Consolidator } from "../../src/rmrk2.0.0";
import { getRemarksFromBlocksMock } from "./mocks";

describe("rmrk2.0.0 Consolidator: CREATE NFT CLASS", () => {
  it("should work", async () => {
    const remarks = getRemarksFromBlocksMock();
    const consolidator = new Consolidator();
    expect(await consolidator.consolidate(remarks)).toMatchSnapshot();
  });
});
