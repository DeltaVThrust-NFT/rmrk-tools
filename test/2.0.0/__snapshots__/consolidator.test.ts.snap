// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rmrk2.0.0 Consolidator: CREATE NFT CLASS should correctly create a NFT Collection 1`] = `
Object {
  "bases": Array [],
  "collections": Array [
    Collection {
      "block": 2,
      "changes": Array [],
      "id": "d43593c715a56da27d-KANARIABIRDS",
      "issuer": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
      "max": 0,
      "metadata": "https://some.url",
      "symbol": "KANARIABIRDS",
    },
  ],
  "invalid": Array [],
  "nfts": Array [],
}
`;
